lib/github.com/mpllang/mpllib/SeqBasis.sml

```diff
@@ -75,6 +81,14 @@ struct
       foldl g b' (lo+1, hi) f
     end
 
+  fun foldr g b (lo, hi) f =
+    if lo >= hi then b else
+    let
+      val b' = g (b, f hi)
```

Comment:
Recommendation: do `f (hi-1)` here. Semantically, we want `(lo, hi)` to represent a range `[lo, hi)`. This would then be consistent with the other functions.

Reply:
See also the comments about `Seq.foldr`. By making this change here, the range needed would `(0, n)` rather than `(~1, n-1)` which is strange.